tests/unit/pluginhandler/test_pluginhandler.py

```diff
@@ -1807,6 +1807,25 @@ def test_clean_prime_state(self):
         self.assertThat(self.handler.next_step(), Equals(steps.PRIME))
         self.assertFalse(os.path.exists(bindir))
 
+    def test_clean_prime_state_inconsistent_files(self):
+        self.assertRaises(errors.NoLatestStepError, self.handler.latest_step)
+        self.assertThat(self.handler.next_step(), Equals(steps.PULL))
+        bindir = os.path.join(self.prime_dir, "bin")
+        os.makedirs(bindir)
+        open(os.path.join(bindir, "1"), "w").close()
+
+        self.handler.mark_done(steps.STAGE)
```

Comment:
in practice, prime is a subset of staged files so mark_done should probably reflect that in this test.
This error could also be triggered by self.handler.clean_stage.